### ISSUE TITLE
Set a default selection for a portfolio

### ISSUE THREAD

**Issue by benel** [Hypertopic/Porphyry#217]
A portfolio could be set so that the initial selection is no more empty but is related to corpora, topics, and maybe attribute values.

For minimal impact on the Hypertopic model, it could be set by admins in the config file. 

It was one of the solutions proposed to reduce initial loading (and display) time.

### Phase 1
- [ ] Scenarios (Gherkin)
- [ ] Mockups
- [ ] Implementation strategy
### Phase 2
- [ ] Acceptance tests (Capybara)
- [ ] Implementation
### Phase 3
- [ ] Screencast

**Comment by Kodlah** [comment]
Projet Graines d'artistes : 
Le portfolio par défaut pourrait être défini pour afficher les dessins qui ont gagné le concours.

**Comment by Starangon** [comment]
Projet Visite sur site de vitraux : 
Ceci pourrait être très pratique pour une personne qui sait où elle veut aller voir un vitrail, elle n'aurait alors pas a charger l’entièreté des vitraux une fois dans la zone avec peu de connexion internet.

**Comment by benel** [comment]
@TheStarTom Pour le besoin que vous exprimez, il suffit d'avoir une URI différente pour chaque item... C'est déjà le cas, par exemple : http://vitraux.porphyry.org/item/Vitraux%20-%20Bénel/8a1750b17b11944108efaac593f4448e4e9f966b
 
Cela permettrait par exemple pour la visite des vitraux d'imprimer un QR code sur une brochure distribuée à l'entrée de l'édifice ou sur un kakemono.

**Comment by benel** [comment]
@Kodlah @moonkeys @ThomasLORIOT Attention si vous implémentez cette fonctionnalité à ce que ça ne casse pas ce qui existe...

Peut-être que dans un premier temps, vous pourriez simplifier les URI du portfolio. Elles étaient très simples au départ et ont été grandement compliquées par vos prédécesseurs quand ils ont implémenté la recherche booléenne.

Avant : `/?t=87c8784de68e67b59bb32b3cb66c7c77`

Après : `/?t={%22type%22:%22intersection%22,%22data%22:[{%22type%22:%22intersection%22,%22selection%22:[%2287c8784de68e67b59bb32b3cb66c7c77%22],%22exclusion%22:[]}]}`

Si les URI redevenaient simples, ce serait plus facile de référencer des présélections dans des pages Web (comme les [lauréats](http://dessins.porphyry.org/?t={%22type%22:%22intersection%22,%22data%22:[{%22type%22:%22intersection%22,%22selection%22:[%223176578e6637823a547a325284d9a897%22],%22exclusion%22:[]}]}) pour le cas de Graines d'artiste).

**Comment by moonkeys** [comment]
Voici la maquette du ticket #217 : 

1. La galerie des lauréats est sélectionnée
2. L'URI a bien été simplifiée


![image](https://user-images.githubusercontent.com/43788230/84238674-b8abc700-aafb-11ea-9dbf-01aceda96420.png)

**Comment by benel** [comment]
> L'URI a bien été simplifiée

Effectivement. Donc j'en conclue que vous avez choisi pour implémenter cette fonctionnalité de simplifier les URI ce qui permettra de référencer des pré-sélections plus facilement. Entendu.

> ![image](https://user-images.githubusercontent.com/597307/84244672-468bb000-ab04-11ea-89af-841a2562f517.png)

Je vois plusieurs incohérences entre votre maquette et votre preuve de concept (confusion entre catégorie et corpus, absence du statut de sélection, présence de drapeaux, position de la recherche, boutons qui ne correspondent à rien, etc. etc.).

Notez que dans ce cas précis, il est beaucoup plus simple et sûr pour réaliser votre maquette de **prendre une copie d'écran** de votre preuve de concept (quand les lauréats sont sélectionnés) et de modifier l'URI.

**Comment by moonkeys** [comment]
Bonjour, 

J'avais repris une mauvaise version de la maquette. J'ai corrigé l'erreur dans le preuve de concept ci-dessous.

![image](https://user-images.githubusercontent.com/43788230/84255317-bb65e680-ab12-11ea-9646-8404c7112d77.png)

**Comment by ThomasLORIOT** [comment]
Voici le scénario du ticket #217 :

Fonctionnalité: mettre une sélection par défaut du portfolio

 Scénario:

   Soit "vitraux" le portfolio ouvert
   Alors la sélection "vitraux Benel" est sélectionné dans le portfolio

**Comment by benel** [comment]
> ![image](https://user-images.githubusercontent.com/597307/84256075-c4a38300-ab13-11ea-88a5-8df7096ca58a.png)

@moonkeys Dans cette copie d'écran : 

- les lauréats sont dans un corpus à part : c'est comme si, en modélisation objet, ils étaient des objets différents de classes différentes, or ça ne correspond pas au cycle de vie du concours (on ne sait pas dès le départ que ce sera un lauréat), si vous regardez [la preuve de concept](http://dessins.porphyry.org) qui correspond aux discussions que l'on a eue tout au long du semestre, il n'y a qu'**un seul corpus**, le fait d'être déclaré comme lauréat survient bien après la création de l'item, par le fait de lui ajouter la catégorie `Nominés > Lauréats`,
- rien n'est sélectionné (la barre de sélection indique `Tous les items`) alors que la fonctionnalité nécessite justement que `Nominés > Lauréats` soit sélectionné (et c'est bien son UUID que vous avez indiqué dans l'URI).

Donc autrement dit : 

1. Dans [votre preuve de concept](http://dessins.porphyry.org), sélectionnez `Nominés > Lauréats`.
2. Prenez une copie d'écran.
3. Modifiez l'URI dans l'image pour être simplifiée.
